Segmentation fault if mysql-monitor_enabled=false

On Alpine I get a segmentation fault if I set mysql-monitor_enabled to false. I'm using v1.4.8. This is the backtrace:
```
(gdb) bt
#0  0x00007fe1620624f9 in ?? () from /lib/ld-musl-x86_64.so.1
#1  0x00005611404380cf in Thread::~Thread (this=0x561141cd9520, __in_chrg=<optimized out>) at thread.cpp:35
#2  0x0000561140436648 in ConsumerThread::~ConsumerThread (this=0x561141cd9520, __in_chrg=<optimized out>) at MySQL_Monitor.cpp:54
#3  0x0000561140436664 in ConsumerThread::~ConsumerThread (this=0x561141cd9520, __in_chrg=<optimized out>) at MySQL_Monitor.cpp:54
#4  0x0000561140430a6b in MySQL_Monitor::run (this=0x7fe16229cc60) at MySQL_Monitor.cpp:1975
#5  0x000056114037ef01 in std::__invoke_impl<void*, void* (MySQL_Monitor::* const&)(), MySQL_Monitor*> (
    __f=@0x561141cd2f30: (void *(MySQL_Monitor::*)(MySQL_Monitor * const)) 0x56114043032a <MySQL_Monitor::run()>, __t=@0x561141cd2f28: 0x7fe16229cc60)
    at /usr/include/c++/6.4.0/functional:227
#6  0x000056114037ee7a in std::__invoke<void* (MySQL_Monitor::* const&)(), MySQL_Monitor*> (
    __fn=@0x561141cd2f30: (void *(MySQL_Monitor::*)(MySQL_Monitor * const)) 0x56114043032a <MySQL_Monitor::run()>, __args#0=@0x561141cd2f28: 0x7fe16229cc60)
    at /usr/include/c++/6.4.0/functional:251
#7  0x000056114037ee1c in std::_Mem_fn_base<void* (MySQL_Monitor::*)(), true>::operator()<MySQL_Monitor*> (this=0x561141cd2f30, 
    __args#0=@0x561141cd2f28: 0x7fe16229cc60) at /usr/include/c++/6.4.0/functional:604
#8  0x000056114037ede7 in std::_Bind_simple<std::_Mem_fn<void* (MySQL_Monitor::*)()> (MySQL_Monitor*)>::_M_invoke<0ul>(std::_Index_tuple<0ul>) (
    this=0x561141cd2f28) at /usr/include/c++/6.4.0/functional:1391
#9  0x000056114037ed22 in std::_Bind_simple<std::_Mem_fn<void* (MySQL_Monitor::*)()> (MySQL_Monitor*)>::operator()() (this=0x561141cd2f28)
    at /usr/include/c++/6.4.0/functional:1380
#10 0x000056114037ecf2 in std::thread::_State_impl<std::_Bind_simple<std::_Mem_fn<void* (MySQL_Monitor::*)()> (MySQL_Monitor*)> >::_M_run() (
    this=0x561141cd2f20) at /usr/include/c++/6.4.0/thread:197
#11 0x00007fe1616f08d1 in std::basic_fstream<char, std::char_traits<char> >::~basic_fstream (this=0x7fe162137bb8, __in_chrg=<optimized out>, 
    __vtt_parm=<optimized out>) at /build/gcc/src/gcc-build/x86_64-pc-linux-gnu/libstdc++-v3/include/fstream:1014
#12 std::basic_fstream<char, std::char_traits<char> >::~basic_fstream (this=0x7fe162137bb8, __in_chrg=<optimized out>, __vtt_parm=<optimized out>)
    at /build/gcc/src/gcc-build/x86_64-pc-linux-gnu/libstdc++-v3/include/fstream:1015
#13 0x0000000000000000 in ?? ()
```

Looking at the code I think `m_running` should be set to `0` instead of `m_detached` [here](https://github.com/sysown/proxysql/blob/669c149f5c0e5ee95dd173ea80bcb4440ab3eda8/lib/thread.cpp#L66).

Btw: I had to remove the block in debug.cpp that uses `backtrace` and `backtrace_symbols` for a debug build because of undefined references:
```
../lib/libproxysql.a(debug.oo): In function `proxy_debug_func':
/tmp/proxysql/lib/debug.cpp:59: undefined reference to `backtrace'
/tmp/proxysql/lib/debug.cpp:61: undefined reference to `backtrace_symbols'
```
Probably this has been forgotten in commit 55b2425e707881c1dc6e238bc45b57cfe7a761f1. 

Oh, and it would be nice if NOJEMALLOC is documented or automatically set.

--- a/lib/thread.cpp
+++ b/lib/thread.cpp
@@ -63,7 +63,7 @@ int Thread::join()
     if (m_running == 1) {
         result = pthread_join(m_tid, NULL);
         if (result == 0) {
-            m_detached = 0;
+            m_running = 0;
         }
     }
     return result;

--- a/lib/thread.cpp
+++ b/lib/thread.cpp
@@ -31,10 +31,10 @@ Thread::Thread() : m_tid(0), m_running(0), m_detached(0) {}
 
 Thread::~Thread()
 {
-    if (m_running == 1 && m_detached == 0) {
-        pthread_detach(m_tid);
-    }
     if (m_running == 1) {
+        if (m_detached == 0) {
+             pthread_detach(m_tid);
+        }
         pthread_cancel(m_tid);
     }
 }

